pythclient/hermes.py

```diff
@@ -0,0 +1,173 @@
+        params = {"ids[]": feedIds, "binary": "true"}
+
+        data = (await self.client.get(url, params=params)).json()
+
+        results = []
+        for res in data:
+            price_feed = self.extract_price_feed(res)
+            results.append(price_feed)
+
+        return results
+
+    async def get_pyth_price_at_time(self, feed_id: str, timestamp: int) -> PriceFeed:
+        """
+        Queries the Hermes https endpoint for the price feed for a Pyth feed ID at a given timestamp.
+        """
+        url = os.path.join(self.endpoint, "get_price_feed")
+        params = {"id": feed_id, "publish_time": timestamp, "binary": "true"}
+
+        data = (await self.client.get(url, params=params)).json()
+
+        price_feed = self.extract_price_feed(data)
+
+        return price_feed
+
+    async def get_all_prices(self) -> dict[str, PriceFeed]:
+        """
+        Queries the Hermes http endpoint for the latest price feeds for all feed IDs in the class object.
+
+        There are limitations on the number of feed IDs that can be queried at once, so this function queries the feed IDs in batches.
+        """
+        pyth_prices_latest = []
+        i = 0
+        batch_size = 100
+        while len(self.feed_ids[i : i + batch_size]) > 0:
+            pyth_prices_latest += await self.get_pyth_prices_latest(
+                self.feed_ids[i : i + batch_size]
+            )
+            i += batch_size
+
+        return dict([(feed['feed_id'], feed) for feed in pyth_prices_latest])
+
+    async def ws_pyth_prices(self):
+        """
+        Opens a websocket connection to Hermes for latest prices for all feed IDs in the class object.
+        """
+        import json
+
+        import websockets
+
+        async with websockets.connect(self.ws_endpoint) as ws:
+            while True:
+                # add new price feed ids to the ws subscription
+                if len(self.pending_feed_ids) > 0:
+                    json_subscribe = {
+                        "ids": self.pending_feed_ids,
+                        "type": "subscribe",
+                        "verbose": True,
+                        "binary": True,
+                    }
+                    await ws.send(json.dumps(json_subscribe))
+                    self.pending_feed_ids = []
+
+                msg = json.loads(await ws.recv())
+                if msg.get("type") == "response":
+                    if msg.get("status") != "success":
+                        raise Exception("Error in subscribing to websocket")
+                try:
+                    if msg["type"] != "price_update":
+                        continue
+
+                    feed_id = msg["price_feed"]["id"]
+                    new_feed = msg["price_feed"]
+
+                    self.prices_dict[feed_id] = self.extract_price_feed(new_feed)
+
+                except:
+                    raise Exception("Error in price_update message", msg)
```

Comment:
it's better to catch specific exceptions or at least log the exception details to make debugging easier 
```
except Exception as e:
            raise Exception(f"Error in price_update message: {msg}") from e
```